src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineLinuxLauncher.java

```diff
@@ -109,6 +118,31 @@ private Optional<Connection> bootstrap(
     return Optional.of(bootstrapConn);
   }
 
+  @Override
+  protected boolean checkStartupScriptFinished(
+      ComputeEngineComputer computer, Connection conn, PrintStream logger, TaskListener listener) {
+    try {
+      // A successful command indicates service is still running
+      if (!testCommand(
+          computer,
+          conn,
+          // default value will be 0 until it finishes
+          "exit $(systemctl show google-startup-scripts --property ExecMainExitTimestampMonotonic | cut -d \"=\" -f 2)",
+          // not "initializing" or "starting" - benefit is doesn't require google-startup-scripts
+          // "[ \"$(systemctl is-system-running)\" != \"starting\" ]",
```

Comment:
Do we think it's better to check that `google-startup-scripts` has finished or that `systemctl is-system-running` does not return "initializing" or "starting"? Those seem to be the two best ways on my ubuntu 18/20 machines, but curious what others think here.

Reply:
Hi @caseyduquettesc thank you. I think it's just fine to check `google-startup-scripts` has finished.

When can this be merged? I was hoping I could use it pretty soon.

Reply:
Will need a project maintainer to review. I'll try asking in the slack channel